src/disk_utils.cpp

```diff
@@ -418,47 +477,194 @@ namespace diskann {
     return 0;
   }
 
+  // TODO: Make this a streaming implementation to avoid exceeding the memory
+  // budget
   template<typename T>
-  int build_merged_vamana_index(std::string     base_file,
-                                diskann::Metric compareMetric, unsigned L,
-                                unsigned R, double sampling_rate,
-                                double ram_budget, std::string mem_index_path,
-                                std::string medoids_file,
-                                std::string centroids_file,
-                                size_t build_pq_bytes, bool use_opq) {
+  void breakup_dense_points(const std::string data_file,
```

Comment:
Add some comments on what this function does? Also, does this need to be in *disk*_utils? Should we create a filter_utils file to consolidate filter-specific helper functions scattered around the folders?

Reply:
@darvg can we add some comments about what this function does?